git-secrets

```diff
@@ -111,18 +111,63 @@ scan_history() {
 git_grep() {
   local options="$1"; shift
   local files=("${@}") combined_patterns=$(load_combined_patterns)
+  local status=0
 
   [ -z "${combined_patterns}" ] && return 1
-  GREP_OPTIONS= LC_ALL=C git grep -nwHEI ${options} "${combined_patterns}" -- "${files[@]}"
+
+  if [ ${#files[@]} -eq 0 ]; then
+    GREP_OPTIONS= LC_ALL=C git grep -nwHEI ${options} "${combined_patterns}"
+    return $?
+  fi
+
+  # let xargs watch for system limit on arg count for us. xargs returns 123 if
+  # any call returned 1, but we care that all calls returned 1, so invert the
+  # output - xargs will return 0 if every call returned 0
+  printf "%s\n" "${files[@]}" |
+      GREP_OPTIONS= LC_ALL=C xargs -P "$(nproc)" -d'\n' sh -c \
```

Comment:
This should probably do the same 0/1/255 trick that regular_grep does, to handle the case where "git grep" e.g. exits with status 128.

Should these have a helper in common to do that?

Reply:
I tried it out, and because there are a variable-length handful of args which need to be on every invocation, plus a variable-length array of args which could be split between invocations, this didn't work out too well. But I did copy the switches here too.